src/lib/utils/logging.ts

```diff
@@ -2,16 +2,69 @@
  * @fileoverview Handle logging for Sonar (based on ESLint)
  */
 
-/* eslint no-console: "off" */
+import * as path from 'path';
+import * as d from 'debug';
 
-/* istanbul ignore next */
+const debugEnabled: boolean = (process.argv.includes('--debug'));
 
-/** Cover for console.error */
-export const error = (message: any, ...optionalParams: Array<any>) => {
-    console.error(message, ...optionalParams);
+// must do this initialization *before* other requires in order to work
+if (debugEnabled) {
+    d.enable('sonar:*');
+}
+
+const getModuleInfo = (filePath: string) => {
+    let output: string = path.basename(filePath, path.extname(filePath));
+    let dirPath: string = path.dirname(filePath);
+    let currentDir: string = path.basename(dirPath);
+
+    // The logger message is generated from the file path, e.g.:
+    //
+    //  * src/lib/collectors/cdp/cdp-launcher.ts => sonar:collectors:cdp:cdp-launcher
+    //  * src/lib/collectors/cdp/cdp.ts => sonar:collectors:cdp
+
+    while (currentDir && currentDir !== 'lib') {
+
+        // If the file is in a directory with the same name, do not add
+        // its parent directory (this is the case for collectors & rules).
+
+        if (currentDir !== output) {
+            output = `${currentDir}:${output}`;
+        }
+
+        dirPath = path.join(dirPath, '..');
+        currentDir = path.basename(dirPath);
+    }
+
+    // For `/src/lib/sonar.ts`, use 'sonar:engine' instead of 'sonar:sonar'.
+    if (output === 'sonar') {
+        output = 'engine';
+    }
+
+    return `sonar:${output}`;
 };
 
-/** Cover for console.log */
-export const log = (message: any, ...optionalParams: Array<any>) => {
-    console.log(message, ...optionalParams);
+export const loggerInitiator = (filePath: string, includesModuleInfo: boolean = true) => {
```

Comment:
We are creating a bunch of this and each time a new `error`, `log` and `debug`. Let's create a class `Logger` and export it so we call it like this:
```js
import { Logger } from './utils/logger'; // Will have to rename the file
const logger = new Logger(__filename);
```